From 6c2054f65ae6beb2a38c6188c0807417adbb3880 Mon Sep 17 00:00:00 2001
From: Jeremy Harris <jgh146exb@wizmail.org>
Date: Thu, 4 Apr 2019 14:33:28 +0100
Subject: [PATCH 16/34] SPF: better buld compatibility with OpenBSD

(cherry picked from commit bda76da8a9357f4fc525b5f8b925fae262c28010)
(cherry picked from commit 804219086fe9afbc1429c309e339524aaaabcec1)
(cherry picked from commit c26e27d5b81ed5640c00ee87f1d4287fb066dc12)
---
 src/lookups/spf.c | 4 ++--
 src/spf.h         | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/lookups/spf.c b/src/lookups/spf.c
index b32a73e6a..48d6ce3b5 100644
--- a/src/lookups/spf.c
+++ b/src/lookups/spf.c
@@ -25,8 +25,8 @@ static void dummy(int x) { dummy2(x-1); }
 #else
 
 #include "lf_functions.h"
-#ifndef HAVE_NS_TYPE
-#define HAVE_NS_TYPE
+#if !defined(HAVE_NS_TYPE) && defined(NS_INADDRSZ)
+# define HAVE_NS_TYPE
 #endif
 #include <spf2/spf.h>
 #include <spf2/spf_dns_resolv.h>
diff --git a/src/spf.h b/src/spf.h
index 23ad325f1..a0779f87c 100644
--- a/src/spf.h
+++ b/src/spf.h
@@ -11,7 +11,7 @@
 #ifdef SUPPORT_SPF
 
 /* Yes, we do have ns_type. spf.h redefines it if we don't set this. Doh */
-#ifndef HAVE_NS_TYPE
+#if !defined(HAVE_NS_TYPE) && defined(NS_INADDRSZ)
 # define HAVE_NS_TYPE
 #endif
 #include <spf2/spf.h>
-- 
2.23.0.rc1

